msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionMessageBean.java

```diff
@@ -449,4 +457,128 @@ public String getAuthorEid(){
 		String userEid = this.getMessage().getCreatedBy();
  		return userEid;
 	}	
+
+	private boolean isAnonymousEnabled()
+	{
+		return ServerConfigurationService.getBoolean("msgcntr.forums.anonymous.enable", true);
+	}
+
+	public void setAnonId(String anonId)
+	{
+		this.anonId = anonId;
+	}
+
+	/**
+	 * Returns the author's anonymousID regardless of whether we're in an anonymous context (unless anonymous is disabled system wide)
+	 */
+	public String getAnonId()
+	{
+		if (!isAnonymousEnabled())
+		{
+			return message.getAuthor();
+		}
+
+		if (anonId != null)
+		{
+			// anonId has already been supplied
+			return anonId;
+		}
+
+		// get / create the anonId from the AnonymousManager
+		AnonymousManager anonymousManager = (AnonymousManager)ComponentManager.get("org.sakaiproject.api.app.messageforums.AnonymousManager");
+		String siteId = ToolManager.getInstance().getCurrentPlacement().getContext();
```

Comment:
There is an effort to replace Covers and remove them; however we shouldn't be adding more work by introducing new instances of them.